internal/services/streamanalytics/stream_analytics_output_cosmosdb_resource.go

```diff
@@ -0,0 +1,297 @@
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, id.ResourceGroup, id.StreamingjobName, id.Name)
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			if props := resp.OutputProperties; props != nil && props.Datasource != nil {
+				v, ok := props.Datasource.AsDocumentDbOutputDataSource()
+				if !ok {
+					return fmt.Errorf("converting output data source to a document DB output: %+v", err)
+				}
+
+				state := OutputCosmosDBResourceModel{
+					Name:               id.Name,
+					StreamAnalyticsJob: id.StreamingJobID(),
+				}
+
+				state.AccountKey = metadata.ResourceData.Get("cosmosdb_account_key").(string)
+
+				if v.Database != nil {
+					state.Database = *v.Database
+				}
+
+				if v.CollectionNamePattern != nil {
+					state.CollectionNamePattern = *v.CollectionNamePattern
+				}
+
+				if v.DocumentID != nil {
+					state.DocumentID = *v.DocumentID
+				}
+
+				if v.PartitionKey != nil {
+					state.PartitionKey = *v.PartitionKey
+				}
+
+				return metadata.Encode(&state)
+			}
+			return nil
+		},
+	}
+}
+
+func (r OutputCosmosDBResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.StreamAnalytics.OutputsClient
+			id, err := parse.OutputID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("deleting %s", *id)
+
+			if resp, err := client.Delete(ctx, id.ResourceGroup, id.StreamingjobName, id.Name); err != nil {
+				if !response.WasNotFound(resp.Response) {
+					return fmt.Errorf("deleting %s: %+v", *id, err)
+				}
+			}
+			return nil
+		},
+	}
+}
+
+func (r OutputCosmosDBResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.OutputID
+}
+
+func (r OutputCosmosDBResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.StreamAnalytics.OutputsClient
+			id, err := parse.OutputID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state OutputCosmosDBResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			if metadata.ResourceData.HasChangesExcept("name", "stream_analytics_job_id") {
+				props := streamanalytics.Output{
+					OutputProperties: &streamanalytics.OutputProperties{
+						Datasource: streamanalytics.DocumentDbOutputDataSource{
+							Type: streamanalytics.TypeBasicOutputDataSourceTypeMicrosoftStorageDocumentDB,
+							DocumentDbOutputDataSourceProperties: &streamanalytics.DocumentDbOutputDataSourceProperties{
+								AccountKey:            &state.AccountKey,
+								Database:              &state.Database,
```

Comment:
Same here, we need to parse this value as an ID and send the name of the database
```suggestion

			databaseId, err := cosmosParse.SqlDatabaseID(state.Database)
			if err != nil {
				return err
			}

			if metadata.ResourceData.HasChangesExcept("name", "stream_analytics_job_id") {
				props := streamanalytics.Output{
					OutputProperties: &streamanalytics.OutputProperties{
						Datasource: streamanalytics.DocumentDbOutputDataSource{
							Type: streamanalytics.TypeBasicOutputDataSourceTypeMicrosoftStorageDocumentDB,
							DocumentDbOutputDataSourceProperties: &streamanalytics.DocumentDbOutputDataSourceProperties{
								AccountKey:            &state.AccountKey,
								Database:              &databaseId.Name,
```